### PR TITLE
Small tweaks to build instructions.

### PR DESCRIPTION
**Pull request by ssk2**
Specifically:
* Tidy up .gitignore to put the `native-*` line at the right level in the hierarchy.
* Update the build instructions to use the latest version of `build.sh` which scripts away many of the manual steps required before.
* Small grammatical tweak to one of the output lines.

Note that I tried using just build_cli.sh and refactoring it to require just one argument but it has a number of dependencies on environment variables that made it more effort than it was worth. Using `build.sh` incurs some overhead (of framework compilation) that we can probably make optional at some point later.

### CODE REVIEW

File: frameworks/template/cli/dcos-template/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 dcos-*
 *.whl
+native-*
```
Comment (nickbp) [inline frameworks/template/cli/dcos-template/.gitignore]: I think every go build directory would need this, not just `helloworld` and `template` CLIs. These are generated by tools/build_go_exe.sh.

- `frameworks/*/cli/*/.gitignore` 
- `sdk/bootstrap/.gitignore`

Comment (ssk2) [inline frameworks/template/cli/dcos-template/.gitignore]: Updated!

**Comment by ssk2** [comment]
@mesosphere-ci retest me please.

**Comment by benclarkwood** [comment]
Not immediately clear to me why this is failing. @ssk2 it appears that it is failing during Kafka install

》installing kafka with service=kafka options={'service': {'placement_constraint': 'hostname:UNIQUE'}} version=None
》installing CLI commands for package 'kafka'
》installing package 'kafka' with service name 'kafka'

This DC/OS Service is currently in preview.
FAILED

**Comment by ssk2** [comment]
I didn't change the build scripts, so not sure why it's failing. Just rebased, let's see if this helps.

**Comment by ssk2** [comment]
@benclarkwood: `build_cli.sh` is still used. Unfortunately this is a little confusing. `build.sh` for a specific framework calls `tools/build_framework.sh` which calls `tools/build_cli.sh`. `tools/build_cli.sh` then calls `tools/build_go_exe.sh` for each operating system (Windows / Darwin/ Linux).

**Comment by ssk2** [comment]
@mesosphere-ci retest this please

**Pull request merged**